src/sql-parser/src/parser.rs

```diff
@@ -3717,6 +3723,86 @@ impl<'a> Parser<'a> {
         ))
     }
 
+    fn parse_create_continual_task_from_retain(&mut self) -> Result<Statement<Raw>, ParserError> {
+        self.expect_keywords(&[CONTINUAL, TASK])?;
+        let name = RawItemName::Name(self.parse_item_name()?);
+        let in_cluster = self.parse_optional_in_cluster()?;
+
+        let with_options = if self.parse_keyword(WITH) {
+            self.expect_token(&Token::LParen)?;
+            let options = self.parse_comma_separated(Parser::parse_continual_task_option)?;
+            self.expect_token(&Token::RParen)?;
+            options
+        } else {
+            vec![]
+        };
```

Comment:
seems like a good change for some de-duplication with `parse_create_continual_task_from_transform(...)` but not blocking